front/src/applications/editor/tools/rangeEdition/speedSection/EditPSLSection.tsx

```diff
@@ -43,16 +50,53 @@ const getNewRSign = (
   } as PSLSign;
 };
 
+const getNewValidSign = (sign: PSLSign, tracks: TrackSectionEntity[]) => {
+  const newSign = cloneDeep(sign);
+  const signTrack = tracks.find((track) => track.properties.id === sign.track);
+  if (signTrack && sign.position > signTrack.properties.length) {
+    return {
+      ...newSign,
+      position: signTrack.properties.length,
+    };
+  }
+  if (sign.position < 0) {
+    return {
+      ...newSign,
+      position: 0,
+    };
+  }
+  return newSign;
+};
+
 const EditPSLSection = ({
   entity,
+  trackSectionsCache,
   setState,
 }: {
   entity: SpeedSectionPslEntity;
+  trackSectionsCache: Record<string, TrackState>;
   setState: (stateOrReducer: PartialOrReducer<RangeEditionState<SpeedSectionEntity>>) => void;
 }) => {
   const { t } = useTranslation();
   const pslExtension = entity.properties.extensions.psl_sncf;
 
+  const signTrackIds = useMemo(() => {
+    const { psl_sncf: pslExtensionSigns } = entity.properties.extensions;
+    const signs = [pslExtensionSigns.z, ...pslExtensionSigns.r, ...pslExtensionSigns.announcement];
+    return removeDuplicates(signs.map((sign) => sign.track));
+  }, [entity.properties.extensions.psl_sncf]);
+
+  const tracks = useMemo(
+    () =>
+      compact(
+        signTrackIds.map((trackId) => {
+          const track = trackSectionsCache[trackId];
+          return track?.type !== 'success' ? null : track.track;
+        })
+      ),
+    [signTrackIds]
+  );
+
```

Comment:
Should you merge these two useMemo in one ? It results with less memoization. 
```suggestion
  const tracks = useMemo(() => {
    const { psl_sncf: pslExtensionSigns } = entity.properties.extensions;
    const signs = [pslExtensionSigns.z, ...pslExtensionSigns.r, ...pslExtensionSigns.announcement];
    const signTrackIds = removeDuplicates(signs.map((sign) => sign.track));
    return compact(
      signTrackIds.map((trackId) => {
        const track = trackSectionsCache[trackId];
        return track?.type !== 'success' ? null : track.track;
      })
    );
  }, [entity.properties.extensions.psl_sncf]);
```

Reply:
I separated into 2 memoized const because otherwise the tracks are recomputed very often and I wanted to reduce the code executed at each modification of the extension, but we can merge it for sure, as you want :)

Reply:
I'm not sure it will work that way. As signTrackIds is an array, it's reference will change each time we change something in the form and tracks will be recomputed anyway. (I'm not 100% sure about the explanation though)
I tried to console.log in both useMemo and both are called each time anything changes in the form.

Reply:
My bad you're right, since it is an array ! I can change that :) Thanks !